### DISCUSSION TITLE
BR with RCP/NCP history and Router or End Device POSIX targets with RCP/NCP?

### DISCUSSION THREAD

**Discussion by jkcko** [openthread/openthread#9860] [Q&A] [answered]
I read that ot-br-posix is only targeting RCPs now but used to work with NCP (on an old branch).  Why is that?  Were there functions does the border router needs to control that the NCP cannot or would not do?  Was this to get away from wpantund?

Are there any Thread Router or End device targets for posix platforms that use either the RCP or NCP?

**Comment by jwhui** [comment, marked as answer]
> I read that ot-br-posix is only targeting RCPs now but used to work with NCP (on an old branch). Why is that?

RCP and NCP are two different ways to implement Thread on a platform that has separate processors for host and 15.4 radio. They have their own tradeoffs. RCP allows for lower cost 15.4 radios by moving more code/memory onto host. NCP allows for lower power by moving more functionality to 15.4 radio, allowing host processor to sleep.

The OpenThread project welcomes support for both RCP and NCP models. However, over the past few years, there has been very little contribution to keep NCP support up-to-date.

> Are there any Thread Router or End device targets for posix platforms that use either the RCP or NCP?

Any 15.4 SoC capable of supporting RCP or NCP mode may be used to implement a Thread Router or End Device.
